src/tests/test_example_diffs.rs

```diff
@@ -3125,4 +3149,30 @@ index 53f98b6..14d6caa 100644
  三æäöø€ÆÄÖ〇Øß三
  ¶
 ";
+
+    const GIT_DIFF_LONG_LINE: &str = r#"\
+commit XXXXXXXXXXXXXXXXXXXXXXXXX
+Author: LUCIANO FURTADO <lrfurtado@gmail.com>
+Date:   Tue Jun 25 14:07:31 2024 -0500
+
+    Ref: https://github.com/dandavison/delta/issues/1760
+
+diff --git a/foobar b/foobar
+index 185a33cd5b..123f893935 100755
+--- a/foobar
++++ b/foobar
+@@ -14,6 +14,11 @@ function UNDERFLOW_EXEC() {
+    XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX XX
+ }
+
++
++
++echo "=== VALIDATE UNDERFLOW"
++/usr/local/bin/check_underflow
++
+ # this is an example of the underflow error
+ echo "=== UNDERFLOW_EXEC"
+ UNDERFLOW_EXEC -i foobar.yaml
+
+"#;
```

Comment:
You can leave out almost all of these +- lines, as you found out the context before the changed lines was the cause.

Reply:
any concerns leaving as is ?

Reply:
Well, it makes the minimal reproducer diff larger than it has to be, and the text sort of indicates that it's these lines. As you have to rebase anyhow to make it build with rust 1.80, maybe you can also trim the added lines down to one or two.